cartographer_ros/cartographer_ros/offline_node_main.cc

```diff
@@ -128,8 +128,12 @@ void Run(std::vector<string> bag_filenames) {
         expected_sensor_ids, options.tracking_frame);
     rosbag::Bag bag;
     bag.open(bag_filename, rosbag::bagmode::Read);
+    rosbag::View view(bag);
 
-    for (const rosbag::MessageInstance& msg : rosbag::View(bag)) {
+    const ::ros::Time begin_time = view.getBeginTime();
+    const ::ros::Time end_time = view.getEndTime();
```

Comment:
Since you never use the `end_time`:
`const ::ros::Duration bag_duration = view.getEndTime() - begin_time;`
here?